server/src/test/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java

```diff
@@ -51,19 +53,21 @@ public void testDanglingIndicesAreRecoveredWhenSettingIsEnabled() throws Excepti
         final Settings settings = buildSettings(true);
         internalCluster().startNodes(3, settings);
 
-        createIndex(INDEX_NAME, Settings.builder().put("number_of_shards", 2).put("number_of_replicas", 2).build());
+        execute("create table test(id integer) clustered into 2 shards with(number_of_replicas = 2)");
 
         // Restart node, deleting the index in its absence, so that there is a dangling index to recover
         internalCluster().restartRandomDataNode(new InternalTestCluster.RestartCallback() {
 
             @Override
             public Settings onNodeStopped(String nodeName) throws Exception {
-                assertAcked(client().admin().indices().prepareDelete(INDEX_NAME));
+                assertAcked(client().admin().indices().prepareDelete(getFqn("test")));
                 return super.onNodeStopped(nodeName);
             }
         });
 
-        assertBusy(() -> assertTrue("Expected dangling index " + INDEX_NAME + " to be recovered", indexExists(INDEX_NAME)));
+        assertBusy(() -> assertThat("Expected dangling index test to be recovered",
+                                    execute("select id from sys.shards where table_name='test'").rowCount(),
```

Comment:
why not use `select table_name information_schema.tables where table_name = 'test'`?